### PR TITLE
Kwp simulation

### PR DESCRIPTION
**Pull request by ankayca**
KWP message frame added to simulation side. MC3360 and arduino uno used for tests. This my first pull request so I anything wrong my bad.

### CODE REVIEW

File: src/OBD9141sim.h [hunk 1]
```diff
@@ -75,7 +75,7 @@ class OBD9141sim{
 
         void handleRequests();
         void answerRequest(uint8_t mode, uint8_t pid);
-
+        bool use_kwp_;
```
Comment (iwanders) [inline src/OBD9141sim.h:78]: Please assign this to be false in the constructor, so next to 
https://github.com/iwanders/OBD9141/blob/838bd4a46490dfa8f70e25217fd4fff337e14fa3/src/OBD9141sim.cpp#L15

File: src/OBD9141sim.cpp [hunk 2]
```diff
@@ -96,6 +102,11 @@ void OBD9141sim::answerRequest(uint8_t mode, uint8_t pid){
     message[3] = 0x40 + mode; // is this actually what happens!?
     message[4] = pid;
     message[5] = 0;
+    if(use_kwp_){
```
Comment (iwanders) [inline src/OBD9141sim.cpp]:
```suggestion
    if (use_kwp_){
```

File: src/OBD9141sim.cpp [hunk 3]
```diff
@@ -96,6 +102,11 @@ void OBD9141sim::answerRequest(uint8_t mode, uint8_t pid){
...
+        // now we modify the message for KWP, the payload is the request_len - 3 header bytes
+        message[0] = 0x82 + (answer->len );
```
Comment (iwanders) [inline src/OBD9141sim.cpp]:
```suggestion
        message[0] = 0x82 + answer->len;
```

File: src/OBD9141sim.cpp [hunk 4]
```diff
@@ -48,6 +48,12 @@ void OBD9141sim::initialize(){
     this->initialised = true;
 }
 
+void OBD9141sim::initializeKWP(){
+    use_kwp_ = true;
```
Comment (iwanders) [inline src/OBD9141sim.cpp]:
```suggestion
    this->use_kwp_ = true;
```

I know it's not necessary, but the other code follows this pattern, so lets keep it similar.

**Review by iwanders — CHANGES_REQUESTED** [review]
Hey @ankayca, thanks for taking the time to file this PR. I'm always pleasantly surprised when the simulator is useful to anyone but me, as I only really wrote it as a development tool for my own projects.

I made a few minor remarks on the code, if you could incorporate those we should be able to merge it.

**Review by ankayca — COMMENTED** [review]
I gonna continue to work with this simulator. Its very useful and there are not too much information about K line on the internet and thanks for your reviews.

File: src/OBD9141sim.h (hunk 1 above)
Comment (iwanders) [reply src/OBD9141sim.h:78]: Please incorporate this feedback, other than that it looks good now, thanks for incorporating the others.

**Comment by iwanders** [comment]
One comment about adding the assignment of `use_kwp_` to false in the constructor wasn't addressed, please add that, other than that it looks good.

**Review by ankayca — COMMENTED** [review]
my bad. assigned

**Comment by iwanders** [comment]
Looks good now, I'm going to merge this.

Thanks for your contribution @ankayca !

**Pull request merged**